src/game_io/game_interactive_player_packet.cc

```diff
@@ -169,6 +173,16 @@ void GameInteractivePlayerPacket::write(FileSystem& fs, Game& game, MapObjectSav
 		}
 	}
 
+	if (iplayer) {
+		fw.unsigned_8(1);
+		iplayer->save_windows(fw, *mos);
+	} else {
+		fw.unsigned_8(0);
+	}
+	// ====================================================
+	// Nothing may be saved after `iplayer->save_windows`!!
+	// ====================================================
```

Comment:
A magic number might happen to appear somewhere among the window data, so that's not possible. Storing the length would be the cleanest way, but for that we'd need to save all windows twice – once to a dry-run functor that counts the bytes written, and once to the file after knowing the length. This sounds undesirable IMHO. I clarified the comment though – any data that will only be loaded by IPlayer *can* safely be saveloaded after this line.

Reply:
Seems fine for now.